devicetypes/sky-nie/min-smart-plug.src/min-smart-plug.groovy

```diff
@@ -56,26 +71,79 @@ metadata {
-                if (it.range) {
-                    input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
-                } else {
-                    input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
-                }
-            }
-        }
+        getConfigParamInput(ledModeParam)
+        getConfigParamInput(autoOffIntervalParam)
+        getConfigParamInput(autoOnIntervalParam)
+        getConfigParamInput(powerFailureRecoveryParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'Paddle Control'is only valid for the devices with product number of MS10ZS, MS12ZS, ZW30, ZW30S, ZW30TS(one of them)",
+                required: false
+        getConfigParamInput(paddleControlParam)
+    }
+}
+
+private getConfigParamInput(param) {
+    if (param.range) {
+        input "configParam${param.num}", "number", title: "${param.name}:", required: false, defaultValue: "${param.value}", range: param.range
+    } else {
+        input "configParam${param.num}", "enum", title: "${param.name}:", required: false, defaultValue: "${param.value}", options: param.options
     }
 }
 
+private addChildButton() {
+    log.warn "Creating Button Device"
+    def child = addChildDevice(
+            "sky-nie",
+            "Child Button",
+            "${device.deviceNetworkId}-2",
+            device.getHub().getId(),
+            [
+                completedSetup: true,
+                isComponent: false,
+                label: "plugButton",
+                componentLabel: "${device.displayName[0..-8]} Button"
+            ]
+    )
+    child?.sendEvent(name:"supportedButtonValues", value:JsonOutput.toJson(["pushed", "down", "down_2x", "up", "up_2x"]), displayed:false)
+    child?.sendEvent(name:"numberOfButtons", value:1, displayed:false)
+    sendButtonEvent("pushed")
+    return child
+}
+
 def installed() {
     logDebug "installed()..."
+    try {
+        def child = addChildButton()
+        child?.sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+    } catch (ex) {
+        log.error("Unable to create button device because the 'Child Button' DTH is not installed",ex)
+    }
     sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
 }
 
+def uninstalled() {
+    logger("debug", "uninstalled()")
+    removeChildButton(childDevices[0])
+}
+
+private removeChildButton(child) {
+    try {
+        log.warn "Removing ${child.displayName}} "
+        deleteChildDevice(child.deviceNetworkId)
+    } catch (ex) {
+        log.error("Unable to remove ${child.displayName}!  Make sure that the device is not being used by any SmartApps.", ex)
+    }
+}
+
```

Comment:
You may remove those functions. Child devices will be removed automatically after the main device is removed.

Reply:
OK! I removed them.